## <center>SUPERNATURAL EMPOWERMENT SUMMIT, '23 </center>

### IMPORT PACKAGES

In [1]:
#Data Handling
import numpy as np
import pandas as pd

#Visualizations
import seaborn as sns
import plotly.express as px
import matplotlib.pyplot as plt

from termcolor import colored

# Feature Processing (Scikit-learn processing, etc. )
from sklearn.preprocessing import LabelEncoder

# Evaluation Metrics
from sklearn.model_selection import train_test_split
from catboost import CatBoostClassifier
from sklearn.metrics import f1_score, recall_score, precision_score, fbeta_score, accuracy_score, roc_auc_score
from sklearn.metrics import confusion_matrix
from sklearn.feature_selection import RFE

#Other packages
import warnings
warnings.filterwarnings('always') 

In [2]:
# Overview of the dataset

ses23 = pd.read_csv("C:/Users/KWABENABOATENG/Desktop/DATA ANALYSIS/PERSONAL/SES 2023/SUPERNATURAL-EMPOWERMENT-SUMMIT---2023/SES '23.csv")
ses23

Timestamp   Title First Name         Last Name  \
0     2023/05/22 7:06:16 PM GMT      Mr      Evans            Coffie   
1     2023/05/22 7:07:14 PM GMT      Mr      Noble              King   
2     2023/05/22 7:07:20 PM GMT      Mr    Appiedu  Owusu Acheampong   
3     2023/05/22 7:08:43 PM GMT    Rev      Albert          T Kamara   
4     2023/05/22 7:08:43 PM GMT    Rev.   David K.             Smith   
...                         ...     ...        ...               ...   
3593  2023/06/13 4:07:53 PM GMT      MR     MARCUS              DELA   
3594  2023/06/13 4:08:47 PM GMT      MR    PATRICK           AMEGAZO   
3595  2023/06/13 4:09:33 PM GMT     REV   EBENEZER         ALIBANWIE   
3596  2023/06/13 4:10:45 PM GMT  PASTOR    RAPHAEL           LAMKPOR   
3597  2023/06/13 4:11:56 PM GMT      MR     DANIEL       OWUSU GYEBI   

     Mobile Number WhatsApp Number                  Email Address Country  \
0       0249884472      0249884472                            NaN   Ghana   
1       0240219533      0240219533                            NaN   Ghana   
2       0244450039      0244450039   appiedu.acheampong@gmail.com   Ghana   
3       0243801382      0243801382                            NaN   Ghana   
4       0245127629      0245127629  rev.davidkwesismith@yahoo.com   Ghana   
...            ...             ...                            ...     ...   
3593    0244021980             NaN                            NaN   GHANA   
3594    0597998224             NaN                            NaN   GHANA   
3595    0203044808             NaN                            NaN   GHANA   
3596    0249415892             NaN                            NaN   GHANA   
3597    0588124577             NaN                            NaN   GHANA   

                           Occupation Name of Church  
0                                 NaN        Teminar  
1                                 NaN   Perez Chapel  
2                                 NaN    Perz Chapel  
3                                 NaN   Perez Chapel  
4     Cleric (Pastors, Reverends etc)          C.Y.N  
...                               ...            ...  
3593                              NaN          PEREZ  
3594                              NaN          PEREZ  
3595                              NaN           MPCI  
3596                              NaN           MPCI  
3597                              NaN          CLWCI  

[3598 rows x 10 columns]

In [3]:
# Check the shape of the dataset.

ses23.shape

(3598, 10)

In [4]:
ses23.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3598 entries, 0 to 3597
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Timestamp        3598 non-null   object
 1   Title            3598 non-null   object
 2   First Name       3598 non-null   object
 3   Last Name        3598 non-null   object
 4   Mobile Number    3598 non-null   object
 5   WhatsApp Number  883 non-null    object
 6   Email Address    1467 non-null   object
 7   Country          3598 non-null   object
 8   Occupation       209 non-null    object
 9   Name of Church   3188 non-null   object
dtypes: object(10)
memory usage: 281.2+ KB


In [5]:
ses23.isnull()

Timestamp  Title  First Name  Last Name  Mobile Number  WhatsApp Number  \
0         False  False       False      False          False            False   
1         False  False       False      False          False            False   
2         False  False       False      False          False            False   
3         False  False       False      False          False            False   
4         False  False       False      False          False            False   
...         ...    ...         ...        ...            ...              ...   
3593      False  False       False      False          False             True   
3594      False  False       False      False          False             True   
3595      False  False       False      False          False             True   
3596      False  False       False      False          False             True   
3597      False  False       False      False          False             True   

      Email Address  Country  Occupation  Name of Church  
0              True    False        True           False  
1              True    False        True           False  
2             False    False        True           False  
3              True    False        True           False  
4             False    False       False           False  
...             ...      ...         ...             ...  
3593           True    False        True           False  
3594           True    False        True           False  
3595           True    False        True           False  
3596           True    False        True           False  
3597           True    False        True           False  

[3598 rows x 10 columns]

In [6]:
ses23.isnull().sum()

Timestamp             0
Title                 0
First Name            0
Last Name             0
Mobile Number         0
WhatsApp Number    2715
Email Address      2131
Country               0
Occupation         3389
Name of Church      410
dtype: int64

In [7]:
# Check for the total null values.

ses23.isnull().sum().sum()

8645

In [8]:
ses23

Timestamp   Title First Name         Last Name  \
0     2023/05/22 7:06:16 PM GMT      Mr      Evans            Coffie   
1     2023/05/22 7:07:14 PM GMT      Mr      Noble              King   
2     2023/05/22 7:07:20 PM GMT      Mr    Appiedu  Owusu Acheampong   
3     2023/05/22 7:08:43 PM GMT    Rev      Albert          T Kamara   
4     2023/05/22 7:08:43 PM GMT    Rev.   David K.             Smith   
...                         ...     ...        ...               ...   
3593  2023/06/13 4:07:53 PM GMT      MR     MARCUS              DELA   
3594  2023/06/13 4:08:47 PM GMT      MR    PATRICK           AMEGAZO   
3595  2023/06/13 4:09:33 PM GMT     REV   EBENEZER         ALIBANWIE   
3596  2023/06/13 4:10:45 PM GMT  PASTOR    RAPHAEL           LAMKPOR   
3597  2023/06/13 4:11:56 PM GMT      MR     DANIEL       OWUSU GYEBI   

     Mobile Number WhatsApp Number                  Email Address Country  \
0       0249884472      0249884472                            NaN   Ghana   
1       0240219533      0240219533                            NaN   Ghana   
2       0244450039      0244450039   appiedu.acheampong@gmail.com   Ghana   
3       0243801382      0243801382                            NaN   Ghana   
4       0245127629      0245127629  rev.davidkwesismith@yahoo.com   Ghana   
...            ...             ...                            ...     ...   
3593    0244021980             NaN                            NaN   GHANA   
3594    0597998224             NaN                            NaN   GHANA   
3595    0203044808             NaN                            NaN   GHANA   
3596    0249415892             NaN                            NaN   GHANA   
3597    0588124577             NaN                            NaN   GHANA   

                           Occupation Name of Church  
0                                 NaN        Teminar  
1                                 NaN   Perez Chapel  
2                                 NaN    Perz Chapel  
3                                 NaN   Perez Chapel  
4     Cleric (Pastors, Reverends etc)          C.Y.N  
...                               ...            ...  
3593                              NaN          PEREZ  
3594                              NaN          PEREZ  
3595                              NaN           MPCI  
3596                              NaN           MPCI  
3597                              NaN          CLWCI  

[3598 rows x 10 columns]

### INSIGHTS

In this data sets, there are;
* 3598 rows & 10 columns.

In [9]:
#Check for missing values in this dataset.

ses23.isna().sum()

Timestamp             0
Title                 0
First Name            0
Last Name             0
Mobile Number         0
WhatsApp Number    2715
Email Address      2131
Country               0
Occupation         3389
Name of Church      410
dtype: int64

#### INSIGHTS

There were missing values in four columns of the dataset.<br>
They are:
* WhatsApp column       - 2715 missing values.
* Email Address column  - 2131 missing values.
* Occupation column     - 3389 missing values.
* Name of Church column - 410 missing values.

In [10]:
ses23.describe(include='all')

Timestamp Title First Name Last Name Mobile Number  \
count                        3598  3598       3598      3598          3598   
unique                       3554   109       1692      2536          3190   
top     2023/05/23 8:01:51 PM GMT    Mr   Emmanuel    Mensah             -   
freq                            2  1035         50        51            14   

       WhatsApp Number Email Address Country                       Occupation  \
count              883          1467    3598                              209   
unique             836          1437      86                                7   
top         0243181458             —   Ghana  Cleric (Pastors, Reverends etc)   
freq                 3             4    2418                               98   

       Name of Church  
count            3188  
unique            763  
top             Perez  
freq              546

### FEATURE ENGINEERING

In [11]:
#Create a copy of the original test dataset

ses23.copy = ses23.copy()

In [12]:
ses23.drop (columns=['Timestamp', 'WhatsApp Number', 'Email Address'], inplace = True)

In [13]:
ses23

Title First Name         Last Name Mobile Number Country  \
0         Mr      Evans            Coffie    0249884472   Ghana   
1         Mr      Noble              King    0240219533   Ghana   
2         Mr    Appiedu  Owusu Acheampong    0244450039   Ghana   
3       Rev      Albert          T Kamara    0243801382   Ghana   
4       Rev.   David K.             Smith    0245127629   Ghana   
...      ...        ...               ...           ...     ...   
3593      MR     MARCUS              DELA    0244021980   GHANA   
3594      MR    PATRICK           AMEGAZO    0597998224   GHANA   
3595     REV   EBENEZER         ALIBANWIE    0203044808   GHANA   
3596  PASTOR    RAPHAEL           LAMKPOR    0249415892   GHANA   
3597      MR     DANIEL       OWUSU GYEBI    0588124577   GHANA   

                           Occupation Name of Church  
0                                 NaN        Teminar  
1                                 NaN   Perez Chapel  
2                                 NaN    Perz Chapel  
3                                 NaN   Perez Chapel  
4     Cleric (Pastors, Reverends etc)          C.Y.N  
...                               ...            ...  
3593                              NaN          PEREZ  
3594                              NaN          PEREZ  
3595                              NaN           MPCI  
3596                              NaN           MPCI  
3597                              NaN          CLWCI  

[3598 rows x 7 columns]

In [14]:
ses23.isnull()

Title  First Name  Last Name  Mobile Number  Country  Occupation  \
0     False       False      False          False    False        True   
1     False       False      False          False    False        True   
2     False       False      False          False    False        True   
3     False       False      False          False    False        True   
4     False       False      False          False    False       False   
...     ...         ...        ...            ...      ...         ...   
3593  False       False      False          False    False        True   
3594  False       False      False          False    False        True   
3595  False       False      False          False    False        True   
3596  False       False      False          False    False        True   
3597  False       False      False          False    False        True   

      Name of Church  
0              False  
1              False  
2              False  
3              False  
4              False  
...              ...  
3593           False  
3594           False  
3595           False  
3596           False  
3597           False  

[3598 rows x 7 columns]

In [15]:
ses23.isnull().sum()

Title                0
First Name           0
Last Name            0
Mobile Number        0
Country              0
Occupation        3389
Name of Church     410
dtype: int64

In [16]:
ses23['Occupation'].fillna('Pastor', inplace=True)

In [17]:
ses23['Name of Church'].fillna('Perez Chapel', inplace=True)

In [18]:
ses23['Name of Church'].unique()

array(['Teminar', 'Perez Chapel', 'Perz Chapel', 'C.Y.N',
       'Perez Chapel, Lome', 'Peniel', 'PEREZ TRASAACO', 'GIPM',
       'Perez Dome', 'Perez ', 'Perez Dome ', 'R.D',
       'Living Hope Internetional', 'Perez', 'L.D.S', 'LGCG', 'PCI',
       'Calvary Power', 'Dome', 'FOLP', 'C.O.P', 'GHARIS', 'Coin Int',
       'PCM', 'TBC', 'AWG', 'Perez Aflao ', 'Cantoments', 'PCI Denu',
       'Perez/French', 'PCI ', 'Perez French ', 'Action Chapel',
       'Christ Lunie du Nation', 'LGC', 'FMBU', 'MK', 'Royalhouse', 'WCI',
       'KCIC', 'Fire family', 'Top Church International ', 'TAC',
       'Ghana ', 'Fiqa', 'Pentecost', 'Perez Amanfro', 'Jesus Power',
       'Anglican', 'KCM', 'VGM', 'HKOMI', '3R', 'Perez- zoeland ', 'EECA',
       'ATOG', 'Havest', 'Posibilities', 'HJM', 'LIBC', 'Perez Zoeland ',
       'Roman', 'Church of Christ', 'Rapha Centre', '-', 'Pci', 'ICGC',
       'CSWC', 'MELN ', 'EFC', 'MELN', 'First Century', 'Deeper Life',
       'TFGM', 'AWC', 'Harvest Trmple', 'Zoela

In [48]:
ses23['Name of Church'] = ses23['Name of Church'].replace('Perz Chapel','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel, Lome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Dome ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Aflao ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Cantoments','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Denu','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez French ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Amanfro','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez- zoeland ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Zoeland ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Zoeland','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCi','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-La','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Zoe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Zealand','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('P C I','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Dome-Above Chapel ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Hq','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Takoradi','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI FRENCH','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI DOME','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci Nkwanta','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/Amsterdam','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Zoeland ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Dome ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci/ French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Campus Ministry','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Campus Ministry ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez DOme','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Zoeland','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Atomic','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelDome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelHq','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelFRENCH','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelZoe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelHq','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelFRENCH','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelDOME','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel/ French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelFrench','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez ChapelNkwanta','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('-','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/French ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Tesano','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Zoe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/ French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez French SCE','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Odorkor','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/ Frecnch','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez/ Dome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Krokobite branch','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Dome French ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Zoe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI - Zoeland ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI H.T','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez chapel ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez K.K.','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez H.Q','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez chapel international -Odorkor','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel international ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Preze','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('P.C.I','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Shiloh Temple - Dawhenya Branch ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel International ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PC1','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chapel Information ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez  Chapel','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez (Spintex)','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Spintex','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Madina','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Odorkor','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez. Adenta','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Zoe Temple','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI (East Legon)','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI-Ofankor ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Not well written ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PC','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Kpando','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Hohoe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Ho','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Pokuase','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Brazzaville','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI UCC','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM UCC','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM KNUST','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI/ French','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Persz','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Spintex branch','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Brazzaville','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('University ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci Dzorwulu','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Gbawe','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Chorkor','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Mallam','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci east legon','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace("K'dua",'Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Manhean','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Afiaman','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Perez Mankesim','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Zion Temple','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ LA','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pci Zoeland','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Adenta Perez ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TESANO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('EGLISE FRANCOPHONE','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI MANKESIM','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ CHAPEL','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('TESANO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('DAWHENYA','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('TESHIE','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PRZ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ FRENCH','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ FRANCOPHONE','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('ZOELAND','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI ZOELAND','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM Winneba','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM UEW','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM, Winneba','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM -KNUST','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM - KNUST','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCM-KNUST','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Not Written','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI MANKESIM','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ CHAPEL','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('DOME','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('GHANA','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('KANDA PCZ','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('KASOA','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace("Perez Cite D'Impact",'Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PCI Lome','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Nsuta Nkwakaw','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('Ghana ','Perez Chapel')


ses23['Name of Church'] = ses23['Name of Church'].replace('C.O.P','Church of Pentecost')

ses23['Name of Church'] = ses23['Name of Church'].replace('Church of Pentecost ','Church of Pentecost')
ses23['Name of Church'] = ses23['Name of Church'].replace('PENTECOST','Church of Pentecost')
ses23['Name of Church'] = ses23['Name of Church'].replace('Pentecost','Church of Pentecost')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')
ses23['Name of Church'] = ses23['Name of Church'].replace('PEREZ TRASAACO','Perez Chapel')


In [49]:
ses23['Name of Church'].unique()

array(['Teminar', 'Perez Chapel', 'C.Y.N', 'Peniel', 'GIPM', 'R.D',
       'Living Hope Internetional', 'L.D.S', 'LGCG', 'Calvary Power',
       'FOLP', 'Church of Pentecost', 'GHARIS', 'Coin Int', 'TBC', 'AWG',
       'Action Chapel', 'Christ Lunie du Nation', 'LGC', 'FMBU', 'MK',
       'Royalhouse', 'WCI', 'KCIC', 'Fire family',
       'Top Church International ', 'TAC', 'Fiqa', 'Jesus Power',
       'Anglican', 'KCM', 'VGM', 'HKOMI', '3R', 'EECA', 'ATOG', 'Havest',
       'Posibilities', 'HJM', 'LIBC', 'Roman', 'Church of Christ',
       'Rapha Centre', 'ICGC', 'CSWC', 'MELN ', 'EFC', 'MELN',
       'First Century', 'Deeper Life', 'TFGM', 'AWC', 'Harvest Trmple',
       'Builders Christian church', 'UBCI', 'Brigade', 'KLM',
       'Torchworld Ministries', 'Possibilities ', 'Assemblies of God',
       'AIG', 'Light of life', 'Blessed', 'MDRC', 'QFC', 'Kiwgiv',
       'Foursquare', 'Fours quare', 'First fruit', 'Shiloh Outreach ',
       'CEE', 'Church of God', 'CPI', 'CPU', 'A.G.A',

In [ ]:
ses23

In [ ]:
ses23['Title'].unique()

In [ ]:
ses23['Title'] = ses23['Title'].replace('Presiding Elderer','Presiding Elder')
ses23['Title'] = ses23['Title'].replace('Rev ','Rev')
ses23['Title'] = ses23['Title'].replace('Pastor','Ps')
ses23['Title'] = ses23['Title'].replace('ELDER MRS.','Elder Mrs')
ses23['Title'] = ses23['Title'].replace('Mrs.','Mrs')
ses23['Title'] = ses23['Title'].replace('Ms','Miss')
ses23['Title'] = ses23['Title'].replace('Ms.','Miss')
ses23['Title'] = ses23['Title'].replace('Mr.','Mr')
ses23['Title'] = ses23['Title'].replace('Mrs ','Mrs')
ses23['Title'] = ses23['Title'].replace('Dr.','Dr')
ses23['Title'] = ses23['Title'].replace('Miss ','Miss')
ses23['Title'] = ses23['Title'].replace('Mr ','Mr')
ses23['Title'] = ses23['Title'].replace('REV ','Rev')
ses23['Title'] = ses23['Title'].replace('REV','Rev')
ses23['Title'] = ses23['Title'].replace('SP','Ps')
ses23['Title'] = ses23['Title'].replace('Mr. ','Mr')
ses23['Title'] = ses23['Title'].replace('Mis','Miss')
ses23['Title'] = ses23['Title'].replace('Rev. ','Rev')
ses23['Title'] = ses23['Title'].replace('Cell leader ','Cell leader')
ses23['Title'] = ses23['Title'].replace('Ms ','Miss')
ses23['Title'] = ses23['Title'].replace('Reverend','Rev')
ses23['Title'] = ses23['Title'].replace('Madam','Mrs')
ses23['Title'] = ses23['Title'].replace('Ps. ','Ps')
ses23['Title'] = ses23['Title'].replace('Bro','Mr')
ses23['Title'] = ses23['Title'].replace('Me','Mr')
ses23['Title'] = ses23['Title'].replace('Ps.','Ps')
ses23['Title'] = ses23['Title'].replace('Sir','Mr')
ses23['Title'] = ses23['Title'].replace('Pastor ','Ps')
ses23['Title'] = ses23['Title'].replace('Deaconess ','Deaconess')
ses23['Title'] = ses23['Title'].replace('Apostle ','Apostle')
ses23['Title'] = ses23['Title'].replace('Madam ','Mrs')
ses23['Title'] = ses23['Title'].replace('Prophet ','Prophet')
ses23['Title'] = ses23['Title'].replace('Evangelist ','Evangelist')
ses23['Title'] = ses23['Title'].replace('Pas')
ses23['Title'] = ses23['Title'].replace('MRS','Mrs')
ses23['Title'] = ses23['Title'].replace('Mrs. ','Mrs')
ses23['Title'] = ses23['Title'].replace('MADAM','Mrs')
ses23['Title'] = ses23['Title'].replace('MR','Mr')
ses23['Title'] = ses23['Title'].replace('PASTOR','Ps')
ses23['Title'] = ses23['Title'].replace('MADAM ','Mrs')
ses23['Title'] = ses23['Title'].replace('Elder ','Elder')
ses23['Title'] = ses23['Title'].replace('Apt', 'Apostle')
ses23['Title'] = ses23['Title'].replace('F','Mr')
ses23['Title'] = ses23['Title'].replace('Asafo Maame','Mrs')
ses23['Title'] = ses23['Title'].replace('Reverend ','Rev')
ses23['Title'] = ses23['Title'].replace('Mr/Ms','Mr')
ses23['Title'] = ses23['Title'].replace('P. Eld.','Presiding Elder')
ses23['Title'] = ses23['Title'].replace('Evang','Evangelist')
ses23['Title'] = ses23['Title'].replace('M & E','Mr')
ses23['Title'] = ses23['Title'].replace('Bishop ','Bishop')
ses23['Title'] = ses23['Title'].replace('Deacon ','Deacon')
ses23['Title'] = ses23['Title'].replace('mrs','Mrs')
ses23['Title'] = ses23['Title'].replace('Daniel','Mr')
ses23['Title'] = ses23['Title'].replace('MISS','Miss')
ses23['Title'] = ses23['Title'].replace('Mrs  ','Mrs')
ses23['Title'] = ses23['Title'].replace('Mes','Mrs')
ses23['Title'] = ses23['Title'].replace('Mt','Mr')
ses23['Title'] = ses23['Title'].replace('Min','Mr')
ses23['Title'] = ses23['Title'].replace('Madame ','Mrs')
ses23['Title'] = ses23['Title'].replace('Mr..','Mr')
ses23['Title'] = ses23['Title'].replace('Pst','Ps')
ses23['Title'] = ses23['Title'].replace('Mre','Mrs')
ses23['Title'] = ses23['Title'].replace('Mr/Miss','Mr')
ses23['Title'] = ses23['Title'].replace('PS','Ps')
ses23['Title'] = ses23['Title'].replace('Apst','Apostle')
ses23['Title'] = ses23['Title'].replace('MRS ','Mrs')
ses23['Title'] = ses23['Title'].replace('PS ','Ps')
ses23['Title'] = ses23['Title'].replace('Mad','Mrs')
ses23['Title'] = ses23['Title'].replace('Deaconnes','Deaconess')
ses23['Title'] = ses23['Title'].replace('Eld','Elder')
ses23['Title'] = ses23['Title'].replace('Overseer','Ps')
ses23['Title'] = ses23['Title'].replace('Brother ','Mr')
ses23['Title'] = ses23['Title'].replace('Mme','Mrs')
ses23['Title'] = ses23['Title'].replace('Evan','Evangelist')
ses23['Title'] = ses23['Title'].replace('Pat ','Ps')
ses23['Title'] = ses23['Title'].replace('Master','Mr')
ses23['Title'] = ses23['Title'].replace('Ps ','Ps')
ses23['Title'] = ses23['Title'].replace('Prb','Prophet')
ses23['Title'] = ses23['Title'].replace('Prb ','Prophet')
ses23['Title'] = ses23['Title'].replace('Madam Awortwe','Mrs')
ses23['Title'] = ses23['Title'].replace('Doctor ','Dr')
ses23['Title'] = ses23['Title'].replace('M','Mr')
ses23['Title'] = ses23['Title'].replace('MR','Mr')
ses23['Title'] = ses23['Title'].replace('mr','Mr')
ses23['Title'] = ses23['Title'].replace('MS','Miss')
ses23['Title'] = ses23['Title'].replace('BISHOP','Bishop')
ses23['Title'] = ses23['Title'].replace('PROPHET','Prophet')
ses23['Title'] = ses23['Title'].replace('M & E ','Mr')
ses23['Title'] = ses23['Title'].replace('Djicopitro','Mr')
ses23['Title'] = ses23['Title'].replace('MR ','Mr')
ses23['Title'] = ses23['Title'].replace('Rev.','Rev')
ses23['Title'] = ses23['Title'].replace('Minister','Mr')
ses23['Title'] = ses23['Title'].replace('Rev. Dr','Rev')
ses23['Title'] = ses23['Title'].replace('Rev. Prophet' ,'Prophet')

In [ ]:
ses23['Title'].unique()

In [ ]:
print("These are the unique titles and their count in this dataset: ")

ses23['Title'].value_counts()

In [ ]:
Total_title = ses23['Title'].value_counts()

# Sort in descending order
Total_title  = Total_title .sort_values(ascending=True)

# Set up the figure and axes
fig, ax = plt.subplots()

# Create a bar plot
ax.bar(Total_title .index, Total_title.values)

# Customize the plot
ax.set_title('TOTAL NUMBER OF TITLE THAT CAME FOR THE SES 2023')
ax.set_xlabel('Title')
ax.set_ylabel('Count')

# Rotate x-axis labels if needed
plt.xticks(rotation=90)

# Display the plot
plt.show()

In [ ]:
ses23['Country'].unique()

In [ ]:
ses23['Country'] = ses23['Country'].replace('GHANA','Ghana')
ses23['Country'] = ses23['Country'].replace('Ghana ','Ghana')
ses23['Country'] = ses23['Country'].replace('Netherlands ','Netherlands')
ses23['Country'] = ses23['Country'].replace('Tchad ','Chad')
ses23['Country'] = ses23['Country'].replace('Tchad','Chad')
ses23['Country'] = ses23['Country'].replace('Ghaba','Ghana')
ses23['Country'] = ses23['Country'].replace('GHANA ','Ghana')
ses23['Country'] = ses23['Country'].replace('Ghans','Ghana')
ses23['Country'] = ses23['Country'].replace('Nigeria ','Nigeria')
ses23['Country'] = ses23['Country'].replace('United Kingdom ','United Kingdom')
ses23['Country'] = ses23['Country'].replace('Togo ','Togo')
ses23['Country'] = ses23['Country'].replace('Cameroon ','Cameroon')
ses23['Country'] = ses23['Country'].replace('Ghanaian ','Ghana')
ses23['Country'] = ses23['Country'].replace('Congo ','Congo')
ses23['Country'] = ses23['Country'].replace('Gabon ','Gabon')
ses23['Country'] = ses23['Country'].replace('Benin ','Benin')
ses23['Country'] = ses23['Country'].replace('Ivory coast','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Lvory coast ','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Coast Ivory','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace("Cote D'Ivoire ",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('C I','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Cote divoire','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('CAR','Central African Republic')
ses23['Country'] = ses23['Country'].replace('Ivory Coast ','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Cole divore ','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Gabon/Ghana','Gabon')
ses23['Country'] = ses23['Country'].replace('R.D.C','Democratic Republic of Congo')
ses23['Country'] = ses23['Country'].replace("Cote D'Ivorie",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Ghana/US','Ghana')
ses23['Country'] = ses23['Country'].replace('South Africa ','South Africa')
ses23['Country'] = ses23['Country'].replace('lawsoneleanor920@gmail.com','Ghana')
ses23['Country'] = ses23['Country'].replace('Cote D’Ivoire','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Cote D’Ivoire ','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('C.A.R','Central African Republic')
ses23['Country'] = ses23['Country'].replace('darkohilda@gmail.com','Ghana')
ses23['Country'] = ses23['Country'].replace("Cóte D'voire",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Cote d ivai','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Ivory Coast','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('T Chad','Chad')
ses23['Country'] = ses23['Country'].replace('UK','United Kingdom')
ses23['Country'] = ses23['Country'].replace('C. A. R','Central African Republic')
ses23['Country'] = ses23['Country'].replace('C.I','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Cote D’Ivore','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Preciouskaledzi77@gmail.com','Ghana')
ses23['Country'] = ses23['Country'].replace('Ghanan','Ghana')
ses23['Country'] = ses23['Country'].replace('C.A.R ','Central African Republic')
ses23['Country'] = ses23['Country'].replace('Central African Republic ','Central African Republic')
ses23['Country'] = ses23['Country'].replace("Cote d' ivior",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('CMR','Cameroon')
ses23['Country'] = ses23['Country'].replace('USA','United States of America')
ses23['Country'] = ses23['Country'].replace('CONGO','Congo')
ses23['Country'] = ses23['Country'].replace('GABON','Gabon')
ses23['Country'] = ses23['Country'].replace('CHAD','Chad')
ses23['Country'] = ses23['Country'].replace('SENEGAL','Senegal')
ses23['Country'] = ses23['Country'].replace('COTE DIVOIRE','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('BURKINA FASO','Burkina Faso')
ses23['Country'] = ses23['Country'].replace('SOUTH AFRICA','South Africa')
ses23['Country'] = ses23['Country'].replace('ghana','Ghana')
ses23['Country'] = ses23['Country'].replace('Guana','Ghana')
ses23['Country'] = ses23['Country'].replace("Cote D'Ivoire",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace("Cote d'Ivoire ",'Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Abijan','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('RCI','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('RCI ','Côté d’Ivoire')
ses23['Country'] = ses23['Country'].replace('Swiss','Switzerland')
ses23['Country'] = ses23['Country'].replace('CHC','Ghana')

In [ ]:
ses23['Country'].unique()

In [ ]:
ses23['Country'].value_counts()

In [ ]:
ses23['Occupation'].unique()

In [ ]:
ses23['Occupation'].value_counts()